new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,42 @@
+import {downloadBlob} from "../utils/file";
+
+
+class LogExportManager {
+    readonly #chunks: string[];
+
+    readonly #numChunks: number;
+
+    readonly #fileName: string;
+
+    constructor (numChunks: number, fileName: string) {
+        this.#chunks = [];
+        this.#numChunks = numChunks;
+        this.#fileName = fileName;
+    }
+
+    getChunkLength () {
+        return this.#chunks.length;
+    }
+
+    appendChunkData (chunkData: string) {
+        if (0 === this.#numChunks) {
+            this.download();
+
+            return;
+        }
+        this.#chunks.push(chunkData);
+        if (this.#chunks.length === this.#numChunks) {
+            this.download();
```

Comment:
Shall we clear the `#chunks` (by setting its length to 1) once the file is downloaded, to save memory?

Reply:
Why not 0?